core/drivers/crypto/caam/caam_jr.c

```diff
@@ -0,0 +1,749 @@
+};
+
+/*
+ * Job Ring module private data
+ */
+struct jr_privdata {
+	vaddr_t baseaddr;               /* Job Ring base address */
+
+	vaddr_t ctrladdr;               /* CAAM virtual base address */
+	paddr_t jroffset;               /* Job Ring address offset */
+	uint64_t paddr_inrings;         /* Physical address of input queue */
+	uint64_t paddr_outrings;        /* Physical address of output queue */
+
+	uint8_t nbJobs;                 /* Number of Job ring entries managed */
+
+	/* Input Job Ring Variables */
+	struct inring_entry *inrings;   /* Input JR HW queue */
+	unsigned int   inlock;          /* Input JR spin lock */
+	uint16_t       inwrite_index;   /* SW Index - next JR entry free */
+
+	/* Output Job Ring Variables */
+	struct outring_entry *outrings; /* Output JR HW queue */
+	unsigned int    outlock;        /* Output JR spin lock */
+	uint16_t        outread_index;  /* SW Index - next JR output done */
+
+	/* Caller Information Variables */
+	struct caller_info *callers;    /* Job Ring Caller information */
+	unsigned int    callers_lock;   /* Job Ring Caller spin lock */
+
+	struct itr_handler it_handler;  /* Interrupt handler */
+};
+
+/*
+ * Job Ring module private data reference
+ */
+static struct jr_privdata *jr_privdata;
+
+/*
+ * Free module resources
+ *
+ * @jr_priv   Reference to the module private data
+ */
+static void do_jr_free(struct jr_privdata *jr_priv)
+{
+	if (jr_priv) {
+		caam_free(jr_priv->inrings);
+		caam_free(jr_priv->outrings);
+		caam_free(jr_priv->callers);
+		caam_free(jr_priv);
+	}
+}
+
+/*
+ * Allocate module resources
+ *
+ * @jr_priv   Reference to the module private data
+ * @nbJobs    Number of jobs to manage in the queue
+ *
+ * Returns:
+ * CAAM_NO_ERROR    Success
+ * CAAM_OUT_MEMORY  Out of memory
+ */
+static enum CAAM_Status do_jr_alloc(struct jr_privdata **privdata,
+					uint8_t nbJobs)
+{
+	enum CAAM_Status retstatus = CAAM_OUT_MEMORY;
+	struct jr_privdata *jr_priv = NULL;
+
+	/* Allocate the Job Ring private data */
+	jr_priv = caam_alloc(sizeof(struct jr_privdata));
+
+	if (!jr_priv) {
+		JR_TRACE("Private Data allocation error");
+		goto end_alloc;
+	}
+
+	/* Setup the number of jobs */
+	jr_priv->nbJobs = nbJobs;
+
+	/* Allocate the input and output job ring queues */
+	jr_priv->inrings  = caam_alloc_align(nbJobs *
+						sizeof(struct inring_entry));
+	jr_priv->outrings = caam_alloc_align(nbJobs *
+						sizeof(struct outring_entry));
+
+	/* Allocate the callers information */
+	jr_priv->callers  = caam_alloc(nbJobs * sizeof(struct caller_info));
+
+	if ((!jr_priv->inrings) || (!jr_priv->outrings) ||
+		(!jr_priv->callers)) {
+		JR_TRACE("JR resources allocation error");
+		goto end_alloc;
+	}
+
+	/* Initialize the spin locks */
+	jr_priv->inlock = SPINLOCK_UNLOCK;
+	jr_priv->outlock = SPINLOCK_UNLOCK;
+	jr_priv->callers_lock = SPINLOCK_UNLOCK;
+
+	/* Initialize the queue counter */
+	jr_priv->inwrite_index = jr_priv->outread_index = 0;
```

Comment:
prefer split over 2 lines:
```
	jr_priv->inwrite_index = 0;
	jr_priv->outread_index = 0;
```
Note that this is not required since `caam_ammoc*()` zeroes allocated memory.

Reply:
ok to split, but I prefer to keep initialization as the caam_alloc is function of a define that could be changed.